### PR TITLE
Add searchbar to home

### PR DESCRIPTION
**Pull request by Whatshisname303**

### CODE REVIEW

**Comment by VicHofs** [comment]
Hey Alex, thanks for the PR! Here are a few issues to fix before merging:
- [ ] Fix the dismissal logic (can't tap out of the toggled state)

- [ ] Include the "glow" styles on toggle:
image

- [ ] Modify the colors from hard-coded to theme-provided values:
image

- [ ] Create the search result components and a prop to pass that data to the searchbar:
image

File: components/Searchbar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    const [inputText, updateInputText] = React.useState("");
+    const inputRef = React.useRef<TextInput>(null);
+
+    const isFocused = inputRef.current?.isFocused();
+
+    return (
+        <Container>
+            <Bar
+                style={{
+                    borderColor: isFocused ? "#7A0019" : "#BC808C",
```
Comment (VicHofs) [inline components/Searchbar/index.tsx:20]: Use theme-provided colors

File: components/Searchbar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                    borderBottomLeftRadius: isFocused ? 0 : 32,
+                    borderBottomRightRadius: isFocused ? 0 : 32,
+                }}
+            >
+                <SearchInput
+                    placeholder="Where to?"
+                    placeholderTextColor="#AB5E6E"
```
Comment (VicHofs) [inline components/Searchbar/index.tsx:27]: Use theme-provided colors

File: components/Searchbar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    return (
+        <Container>
+            <Bar
+                style={{
+                    borderColor: isFocused ? "#7A0019" : "#BC808C",
+                    borderBottomLeftRadius: isFocused ? 0 : 32,
+                    borderBottomRightRadius: isFocused ? 0 : 32,
+                }}
+            >
+                <SearchInput
```
Comment (VicHofs) [inline components/Searchbar/index.tsx:25]: Set caret color to a theme color instead of the default

File: components/Searchbar/index.tsx (hunk 1 above)
Comment (VicHofs) [inline components/Searchbar/index.tsx:20]: Use theme-provided colors

File: components/Searchbar/styles.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+// export const Container = styled.View`
+//     position: absolute;
+//     top: 0;
+//     left: 0;
+//     width: 200px;
+//     bottom: 0;
+//     background-color: red;
+// `
+
+export const Container = styled.View`
```
Comment (VicHofs) [inline components/Searchbar/styles.ts:12]: Don't position this absolutely yourself! There is a `Content` component in `Home` that exists precisely for the purpose of absolutely-positioning these overlay elements over the map view.

**Review by VicHofs — CHANGES_REQUESTED** [review]
Hey Alex, thanks for the PR! Here are a few issues to fix before merging:
- [x] Fix the dismissal logic (can't tap out of the toggled state)

- [x] Include the "glow" styles on toggle:
image

- [x] Modify the colors from hard-coded to theme-provided values:
image

- [ ] Create the search result components and a prop to pass that data to the searchbar:
image

- [ ] Change the color of the caret in the input from the default to a theme color:
image

**Comment by Whatshisname303** [comment]
### Shadows
Android and IOS have different ways of managing shadows that I tried to mash together in the CSS. For some reason custom colors didn't work, and I couldn't get it to look right in general. I also don't have any way of testing IOS so would be good to have someone else look at that.

Found a package `react-native-drop-shadow` although it makes images and says it's harder on resources so I didn't decide to use it yet.

### Dismissal logic
I couldn't figure out how to get the search-bar to lose focus while tapping outside the search-bar without making a view cover the rest of the screen which I read is a bad idea, although it works if you're tapping on some other element.

I don't know exactly what actions should dismiss the focus so I just added in the obvious ones that worked easily for now
- Keyboard back button
- Keyboard submit button
- Tapping something in the dropdown

### Other
There were some colors I couldn't find in the theme so I left them hardcoded
- unfocused search-bar border: #bc808c 
- shadow color: #ffe9b9 

Search results are hard coded template data right now rather than props. Also seems like it might be better to handle it all within the component rather than passing anything around.

**Review by VicHofs — CHANGES_REQUESTED** [review]
Hey Alex, thanks for reviewing your PR! There's still a few things left to fix before merging, summarized below:

### Shadows
You can do this without packages; look at [this article](https://blog.logrocket.com/applying-box-shadows-in-react-native/) for reference. Alternatively, feel free to use any libraries you deem adequate!

### Dismissal logic
There is already a View on top of everything; it's the `Content` component I mentioned before. Currently, it has a prop set to ignore all taps, but you can also use a different value and pass-through taps to the map while still reacting to that same event. The dismissal is fine as-is right now, though. **Do make sure the results container is not visible if there are no results to display.**

### Other
- [ ] Move the `ScrollView` inside the View containing the results; you can scroll the whole search bar out of view at the moment
- [x] Add the missing magnifying glass icon on the left
- [ ] Add a 'tapped' state to the search result components
- [x] Add missing building icons in the search result components

_See more comments in the code review!_

File: components/Searchbar/index.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    return (
+        <Container>
+            <ScrollView>
+                <Bar
+                    style={{
+                        borderColor: isFocused ? theme.colors.primaryMain : "#BC808C",
+                        borderBottomLeftRadius: isFocused ? 0 : 32,
+                        borderBottomRightRadius: isFocused ? 0 : 32,
+                        shadowColor: isFocused ? "#FFE9B9" : "rgba(0,0,0,0)",
```
[the remark this answers is not in the document]
Comment (VicHofs) [reply components/Searchbar/index.tsx:38]: If your reference is using colors outside the theme, use the most similar color to each in the theme instead

File: components/Searchbar/index.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
+import React, { useEffect } from "react";
+import { Keyboard, ScrollView, TextInput } from "react-native";
+import { Bar, BuildingAddress, BuildingButton, BuildingName, Container, SearchInput, SearchResultContainer } from "./styles";
+import { useTheme } from "styled-components/native";
+
+const TEMP_BUILDINGS = [
```
[the remark this answers is not in the document]
Comment (VicHofs) [reply components/Searchbar/index.tsx:6]: You can leave this data commented somewhere if you'd like to make it easy to test, though

**Comment by Whatshisname303** [comment]
- Still couldn't figure out shadows
- Don't know what adding tapped state to search result means
- Still got an underline passing refs to styled components

Things should otherwise be working

**Pull request merged**